### ISSUE TITLE
Spy displays nothing for full arrays

### ISSUE THREAD

**Issue by payoto** [matplotlib/matplotlib#18917]
### Bug report

**Bug summary**

`pyplot.spy` and `Axes.spy` display nothing when an array has no 0 values.

**Code for reproduction**

```python
fig, axs = plt.subplots(1, 3)
axs[0].set_title("ones")
axs[0].spy([[1, 1],[1, 1],])
axs[1].set_title("zeros")
axs[1].spy([[0, 0],[0, 0],])
axs[2].set_title("mixed")
axs[2].spy([[0, 1],[1, 0],])
```

**Actual outcome**

![image](https://user-images.githubusercontent.com/18074599/98563189-ca5beb00-22a2-11eb-868a-4e6ff992f941.png)

**Expected outcome**

I would expect the first subplot on the left to be fully black.

**Matplotlib version**

  * Operating system: Windows 10
  * Matplotlib version:  3.3.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): module://ipykernel.pylab.backend_inline
  * Python version:  3.8.6
  * Jupyter version (if applicable):  1.0.0
  * Other libraries:  Many

Installed with conda from conda-forge

**Comment by jklymak** [comment]
... whether we should have such a function or not, this is a pretty easy fix.

**Comment by tacaswell** [comment]
The source of the problem is that we are not setting the vmin/vmax when we call `imshow` under the good.  If all the values are equal we end up inferring that the vmin == vmax == 1.  In the case of equal top / bottom values in color mapping we pick the bottom value (in the case of all empty "does the right thing") hence the all white output.

The fix is to add a check like the one one the interpolation kwarg in https://github.com/matplotlib/matplotlib/blob/bba5c765f750988fe2b01f07babb74cac17d9e8f/lib/matplotlib/axes/_axes.py#L7661-L7665 for vmin, vmax, and norm and then always pass `norm=NoNorm()` to the imshow.


-----

We have had `spy` from at least 2008, independent of if we would merge it now, deprecating it is a non-starter.

**Comment by reshamas** [comment]
I would like to work on this one.

**Comment by jklymak** [comment]
Great.   Please read the developer guide and feel free to ask any questions.

**Comment by reshamas** [comment]
I am no longer working on this issue, so it's available if someone else wants to take it.

**Comment by kristian-cho** [comment]
I will be working on this.

**Comment by bakesbasha** [comment]
Me and @kristian-cho added a fix #18983
